pkg/lspext/partial.go

```diff
@@ -0,0 +1,15 @@
+package lspext
+
+import "github.com/sourcegraph/go-langserver/pkg/lsp"
+
+// PartialResultParams is the input for "$/partialResult", a notification.
+type PartialResultParams struct {
+	ID      lsp.ID      `json:"id"`
+	Patches []JSONPatch `json:"patches"`
+}
+
+type JSONPatch struct {
```

Comment:
Our current use case is super simple, we don't need patching algorithms in the go-langserver.